tests/tst-reloc.c

```diff
@@ -0,0 +1,28 @@
+#include <stdio.h>
+
+int func_impl()
+{
+   return 42;
+}
+
+void *func_resolver() {
+    puts("func_resolver");
+    return (void *)func_impl;
+}
+
+int func() __attribute__((ifunc("func_resolver")));
+
+// .rela.dyn.rel => R_X86_64_64 referencing STT_GNU_IFUNC in .rela.dyn
+int (*fptr_func)() = func;
+
+static void test_indirect_relocations()
+{
+#ifdef __x86_64__
+    printf("%d\n", func());
```

Comment:
nitpick: you could have tested that func() returns 42.
But I admit that any reasonable failure mode would have this test crashing so even without testing anything, it's a good-enough test.